src/codegen/llvm.zig

```diff
@@ -4761,19 +4761,23 @@ fn llvmFieldIndex(
     if (struct_obj.layout != .Packed) {
         var llvm_field_index: c_uint = 0;
         for (struct_obj.fields.values()) |field, i| {
-            if (!field.ty.hasCodeGenBits()) continue;
-
-            if (i == field_index) {
-                ptr_pl_buf.* = .{
-                    .data = .{
-                        .pointee_type = field.ty,
-                        .@"align" = field.normalAlignment(target),
-                        .@"addrspace" = .generic,
-                    },
-                };
-                return llvm_field_index;
+            if (i != field_index) {
+                if (field.ty.hasCodeGenBits())
+                    llvm_field_index += 1;
+                continue;
             }
-            llvm_field_index += 1;
+
+            ptr_pl_buf.* = .{
+                .data = .{
+                    .pointee_type = if (field.ty.hasCodeGenBits())
+                        field.ty
+                    else
+                        Type.initTag(.i8),
+                    .@"align" = field.normalAlignment(target),
+                    .@"addrspace" = .generic,
+                },
+            };
+            return llvm_field_index;
```

Comment:
Actually, what are the semantics of taking a pointer to zero sized fields? Right now, I'm doing a pointer to the next none zero sized field, but it might as well just be a pointer to the start of the struct. I guess the semantics are "a pointer to an unspecified offset into the struct" so we can choose whatever. Maybe a pointer to the start of the struct is less buggy then, idk :shrug:

Reply:
Wait, no, this gives you a pointer to the previous none zero sized field. Truly, I understand my own code well.

Reply:
In stage2, `@sizeOf(void) == @sizeOf(i32)` because it implements  #6706